apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/dao/state/exchange/ExchangeTable.java

```diff
@@ -0,0 +1,128 @@
+/*
+ *  Copyright © 2018-2020 Apollo Foundation
+ */
+
+package com.apollocurrency.aplwallet.apl.core.dao.state.exchange;
+
+import javax.inject.Singleton;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.apollocurrency.aplwallet.apl.core.db.DbIterator;
+import com.apollocurrency.aplwallet.apl.core.db.DbKey;
+import com.apollocurrency.aplwallet.apl.core.db.DbUtils;
+import com.apollocurrency.aplwallet.apl.core.db.LinkKey;
+import com.apollocurrency.aplwallet.apl.core.db.LinkKeyFactory;
+import com.apollocurrency.aplwallet.apl.core.db.TransactionalDataSource;
+import com.apollocurrency.aplwallet.apl.core.db.derived.EntityDbTable;
+import com.apollocurrency.aplwallet.apl.core.entity.state.exchange.Exchange;
+
+@Singleton
+public class ExchangeTable extends EntityDbTable<Exchange> {
```

Comment:
I'm not sure in that, because it operates on it's own 'exchange' table/entity and that seems not related neither 'currency' nor 'asset', if I'm not mistaken.